src/frame/js/actions/faye.js

```diff
@@ -143,9 +144,30 @@ function handleActivityEvents(events) {
 }
 
 function updateUser(currentAppUser, nextAppUser, client) {
-    return (dispatch) => {
+    return (dispatch, getState) => {
         if (currentAppUser._id !== nextAppUser._id) {
-            // wat do?
+            const {config: {appId}} = getState();
+            const actions = [];
+            actions.push(setUser({
+                ...currentAppUser,
```

Comment:
At first I was keeping the `currentAppUser` props because some views were exploding from a lack of `clients` and `pendingClients` array, but I fixed that in a later PR, so in fact yes we might be able to remove this